pkg/util/option_values.go

```diff
@@ -0,0 +1,164 @@
+
+// OptionValuesFromList returns a new OptionValues object from the provided
+// list.
+func OptionValuesFromList(in []vimtypes.BaseOptionValue) OptionValues {
+	if len(in) == 0 {
+		return nil
+	}
+	out := make(OptionValues, len(in))
+	for i := range in {
+		if optVal := in[i].GetOptionValue(); optVal != nil {
+			out[optVal.Key] = getOptionValue(*optVal)
+		}
+	}
+	return out
+}
+
+// ToList returns a list of vimtypes.BaseOptionValue objects.
+func (ov OptionValues) ToList() []vimtypes.BaseOptionValue {
+	if len(ov) == 0 {
+		return nil
+	}
+	i := 0
+	out := make([]vimtypes.BaseOptionValue, len(ov))
+	for k, v := range ov {
+		out[i] = &vimtypes.OptionValue{Key: k, Value: v}
+		i++
+	}
+	return out
+}
+
+// Diff returns the difference between the existing data and provided input.
+// Please note the output does not maintain the order of the input.
+func (ov OptionValues) Diff(in OptionValues) OptionValues {
+	out := OptionValues{}
+	for k, newVal := range in {
+		if oldVal, ok := ov[k]; !ok || newVal != oldVal {
+			out[k] = newVal
+		}
+	}
+	if len(out) == 0 {
+		return nil
+	}
+	return out
+}
+
+// DiffList returns the difference between the existing data and provided input.
+// Please note the output does not maintain the order of the input.
+func (ov OptionValues) DiffList(in []vimtypes.BaseOptionValue) OptionValues {
+	out := OptionValues{}
+	for i := range in {
+		if optVal := in[i].GetOptionValue(); optVal != nil {
+			k := optVal.Key
+			newVal := getOptionValue(*optVal)
+			if oldVal, ok := ov[k]; !ok || newVal != oldVal {
+				out[k] = newVal
+			}
+		}
+	}
+	if len(out) == 0 {
+		return nil
+	}
+	return out
+}
+
+// DiffListToList returns the difference between the existing data and provided
+// input. Please note the output retains the order of the input.
+func (ov OptionValues) DiffListToList(in []vimtypes.BaseOptionValue) []vimtypes.BaseOptionValue {
+	out := []vimtypes.BaseOptionValue{}
+	for i := range in {
+		if optVal := in[i].GetOptionValue(); optVal != nil {
+			k := optVal.Key
+			newVal := getOptionValue(*optVal)
+			if oldVal, ok := ov[k]; !ok || newVal != oldVal {
+				out = append(out, &vimtypes.OptionValue{Key: k, Value: newVal})
+			}
+		}
+	}
+	if len(out) == 0 {
+		return nil
+	}
+	return out
+}
+
+// Append combines the provided input with the current OptionValues data,
+// ignoring any keys from the input data that already exist in the existing
+// data.
+func (ov OptionValues) Append(in OptionValues) {
+	ov.merge(in, false)
+}
+
+// AppendList combines the provided input with the current OptionValues data,
+// ignoring any keys from the input data that already exist in the existing
+// data.
+func (ov OptionValues) AppendList(in []vimtypes.BaseOptionValue) {
+	ov.mergeList(in, false)
+}
+
+// Merge combines the provided input with the current OptionValues data. Any
+// matching keys from the input data will overwrite keys in the existing data.
+func (ov OptionValues) Merge(in OptionValues) {
```

Comment:
I don't see this one being called anywhere as well.  If this is just for future proofing, I would vote for adding it when it's needed.  Sorry if I missed where it is referenced in the diff.

Reply:
The old function was called `MergeExtraConfig`, but it was improperly named. It should have been called `DiffExtraConfig`. I am keeping the `Merge` here in case someone _does_ need it in the future to ensure it is written the correct way.